src/main/java/org/testng/xml/Parser.java

```diff
@@ -248,7 +248,11 @@ private static IFileParser getParser(String fileName) {
    * @return - <code>true</code> if the uri has "file:" as its scheme.
    */
   public static boolean hasFileScheme(String uri) {
-    String scheme = URI.create(uri).getScheme();
+    URI constructedURI = constructURI(uri);
+    if (constructedURI == null) {
+      constructedURI = new File(uri).toURI();
```

Comment:
Will always start by "file", so you can directly return true

Reply:
Fixed.